src/control/cmd/dmg/main.go

```diff
@@ -216,24 +223,48 @@ func parseOpts(args []string, opts *cliOptions, ctlClient control.Invoker, conns
 			logCmd.setLog(log)
 		}
 
-		if opts.ConfigPath == "" {
-			defaultConfigPath := path.Join(build.ConfigDir, defaultConfigFile)
-			if _, err := os.Stat(defaultConfigPath); err == nil {
-				opts.ConfigPath = defaultConfigPath
+		ctlCfg, err := control.LoadConfig(opts.ConfigPath)
+		if err != nil {
+			if opts.ConfigPath != "" {
+				return errors.WithMessage(err, "failed to load control configuration")
 			}
+			ctlCfg = control.DefaultConfig()
+		}
+		if opts.ConfigPath != "" {
+			log.Debugf("control config loaded from %s", opts.ConfigPath)
 		}
 
-		config, err := client.GetConfig(log, opts.ConfigPath)
-		if err != nil {
-			return errors.WithMessage(err, "processing config file")
+		if opts.Insecure {
+			ctlCfg.TransportConfig.AllowInsecure = true
+		}
+		if err := ctlCfg.TransportConfig.PreLoadCertData(); err != nil {
+			return errors.Wrap(err, "Unable to load Certificate Data")
 		}
 
-		ctlClientCfg, err := control.LoadClientConfig(opts.ConfigPath)
-		if err != nil {
-			if opts.ConfigPath != "" {
-				return errors.WithMessage(err, "failed to load client configuration")
+		invoker.SetConfig(ctlCfg)
+		if ctlCmd, ok := cmd.(ctlInvoker); ok {
+			ctlCmd.setInvoker(invoker)
+			if opts.HostList != "" {
+				if hlCmd, ok := cmd.(hostListSetter); ok {
+					hlCmd.setHostList(strings.Split(opts.HostList, ","))
+				} else {
+					return errors.Errorf("MS commands do not accept a hostlist parameter (set it in %s or %s)",
```

Comment:
I'm planning on updating dmg such that the hostlist argument is only available on commands for which it's valid, instead of being a global argument. When that happens, this message will be removed. In the interim, I figured that it could help people migrate to the new way of doing things.

Reply:
Decided to merge master and try for a cleaner test run, so updated the message.